scripts/bootstrap.sh

```
@@ -201,6 +209,35 @@ while [[ "$#" -gt 0 ]]; do
+if [[ -z "$BACKEND" || -z "$EMBEDDING_PROVIDER" ]]; then
+    if [ "$FORCE_DEFAULTS" = "true" ]; then
+        BACKEND=${BACKEND:-"sqlite"}
+        EMBEDDING_PROVIDER=${EMBEDDING_PROVIDER:-"local-model"}
+    elif [ -t 0 ] && [ -t 1 ]; then
+        echo -e "${BLUE}Required options (--backend and --embedding) are missing.${NC}"
+        echo "Default settings will be used:"
+        echo "  Backend: sqlite"
+        echo "  Embedding: local-model"
+        echo ""
+        # Ask for confirmation with a 10-second timeout to prevent hangs in pseudo-TTY agent execution
+        read -t 10 -p "Do you want to proceed with default settings? [y/N]: " CONFIRM || CONFIRM="n"
```

Shellcheck (0.11.0)

[info] read without -r will mangle backslashes.

(SC2162)